modules/features2d/test/test_sift.cpp

```diff
@@ -0,0 +1,145 @@
+/*M///////////////////////////////////////////////////////////////////////////////////////
+//
+//  IMPORTANT: READ BEFORE DOWNLOADING, COPYING, INSTALLING OR USING.
+//
+//  By downloading, copying, installing or using the software you agree to this license.
+//  If you do not agree to this license, do not download, install,
+//  copy or use the software.
+//
+//
+//                           License Agreement
+//                For Open Source Computer Vision Library
+//
+// Copyright (C) 2000-2008, Intel Corporation, all rights reserved.
+// Copyright (C) 2009, Willow Garage Inc., all rights reserved.
+// Third party copyrights are property of their respective owners.
+//
+// Redistribution and use in source and binary forms, with or without modification,
+// are permitted provided that the following conditions are met:
+//
+//   * Redistribution's of source code must retain the above copyright notice,
+//     this list of conditions and the following disclaimer.
+//
+//   * Redistribution's in binary form must reproduce the above copyright notice,
+//     this list of conditions and the following disclaimer in the documentation
+//     and/or other materials provided with the distribution.
+//
+//   * The name of the copyright holders may not be used to endorse or promote products
+//     derived from this software without specific prior written permission.
+//
+// This software is provided by the copyright holders and contributors "as is" and
+// any express or implied warranties, including, but not limited to, the implied
+// warranties of merchantability and fitness for a particular purpose are disclaimed.
+// In no event shall the Intel Corporation or contributors be liable for any direct,
+// indirect, incidental, special, exemplary, or consequential damages
+// (including, but not limited to, procurement of substitute goods or services;
+// loss of use, data, or profits; or business interruption) however caused
+// and on any theory of liability, whether in contract, strict liability,
+// or tort (including negligence or otherwise) arising in any way out of
+// the use of this software, even if advised of the possibility of such damage.
+//
+//M*/
+
+#include "test_precomp.hpp"
+
+namespace opencv_test { namespace {
+
+class CV_SiftTest : public cvtest::BaseTest
+{
+public:
+    CV_SiftTest();
+    ~CV_SiftTest();
+protected:
+    void run(int);
+};
+
+CV_SiftTest::CV_SiftTest() {}
+CV_SiftTest::~CV_SiftTest() {}
+
+void CV_SiftTest::run( int )
```

Comment:
Sorry for confusing you about regression test for SIFT descriptor. It looks like there is [one](https://github.com/terfendail/opencv/blob/fdb8b3dfa887738f48d742f451b01391066ea2d9/modules/features2d/test/test_descriptors_regression.cpp#L21). It verifies against `opencv_extra/testdata/cv/features2d/descriptor_extractors/descriptor-sift` file.
Could you please check it? If there already is a test against precomputed values additional test becomes redundant.

Reply:
This test case will be passed when 80% of results are in an allowable error.
In contrast commit fdb8b3d is for testing bit-exactness and will be passed only when all results are exactly similar.
So I think we will need this test in implementing bit-exact version but I don't think it is necessary now.

Reply:
I've missed the test threshold. In that case exact test is necessary for future development.